pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -760,6 +761,18 @@ func newCloud(config io.Reader) (*Cloud, error) {
 		AssumeRoleTokenProvider: stscreds.StdinTokenProvider,
 	}))
 
+	// If region has not been provided, let's try to get it from an EC2
+	// metadata service and fail if we cannot get that way.
+	if *sess.Config.Region == "" {
+		s := session.Must(session.NewSession(aws.NewConfig().WithMaxRetries(0)))
+		m := ec2metadata.New(s)
+		r, err := m.Region()
+		if err != nil {
+			return &Cloud{}, errors.New("unable to determine region")
+		}
+		sess.Config.Region = &r
```

Comment:
(just thinking out aloud) It's useful to know if/when this kind of "magic" fallback has been applied. Some potential approaches:
- If the user doesn't really need to know as it happens, just document it somewhere more visible
- If the user would like to know when it happens, could be good to log to console (or wherever) – does mean the method would need to codify this somehow – one approach is to set a different field, e.g. `sess.Config.FallbackRegion`, so the consumer of this method can take necessary action. However this is only really needed IF the user needs to know when this happens.